hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -396,32 +402,34 @@ void notifyNormalResponse(Object value, int expectedBackups) {
         future.complete(value);
     }
 
-    @SuppressFBWarnings(value = "VO_VOLATILE_INCREMENT",
-            justification = "We have the guarantee that only a single thread at any given time can change the volatile field")
     void notifyCallTimeout() {
-        operationService.callTimeoutCount.inc();
-
         if (logger.isFinestEnabled()) {
             logger.finest("Call timed-out either in operation queue or during wait-notify phase, retrying call: "
                     + toString());
         }
 
-        if (op instanceof BlockingOperation) {
-            // decrement wait-timeout by call-timeout
-            long waitTimeout = op.getWaitTimeout();
-            waitTimeout -= callTimeout;
-            op.setWaitTimeout(waitTimeout);
+        if (!(op instanceof BlockingOperation)) {
```

Comment:
Correct. With the old behavior the operation is retried after a call get rejected because it failed to start within the given time window. In the new approach it is end of story; as soon as the remote side has rejected your call because it was too late; it isn't retried. It simplifies the flow + makes core easier to test